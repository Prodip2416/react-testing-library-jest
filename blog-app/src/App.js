import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const [data,setData] = useState('');
  return (
    <div className="App">
      {/* <p>Learning Unit testing by react testing library Jest</p>
      <p>JEST.JS</p>
      <img src="" title='hello jest' />
      <input type='text' name="username" placeholder='Enter UserName .... ' onChange={(e)=>setData(e.target.value)}/> */}

      <button onClick={()=>setData('Updated Data')}>Update Data</button>
      <p>{data}</p>
    </div>
  );
}

export default App;
