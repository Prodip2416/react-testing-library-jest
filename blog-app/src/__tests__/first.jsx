import { fireEvent, render, screen } from "@testing-library/react"
import App from "../App"


// test("first test check",()=>{

//   render(<App/>);
//   const firstText = screen.getByText(
//     "Learning Unit testing by react testing library Jest"
//   );
//   const secondText = screen.getByText(/jest.js/i);
//   const firstTitle = screen.getByTitle(/Hello jesT/i);

//   expect(firstText).toBeInTheDocument();
//   expect(secondText).toBeInTheDocument();
//   expect(firstTitle).toBeInTheDocument();
// });

// test("input checking:",()=>{

//   render(<App/>);
//   const inputCheck = screen.getByRole("textbox");
//   const inputPlaceholderText = screen.getByPlaceholderText("Enter UserName ....");

//   expect(inputCheck).toBeInTheDocument();
//   expect(inputCheck).toHaveAttribute("name","username");
//   expect(inputCheck).toHaveAttribute("type","text");
//   expect(inputPlaceholderText).toBeInTheDocument();
// });

// test('onchange render',()=>{
//   render(<App/>);
//   let input = screen.getByRole('textbox');
//   fireEvent.change(input, {target:{value:'a'}});

//   expect(input.value).toBe("a");
// });

// beforeAll(()=>{
//   console.log('before all calling!')
// });
// beforeEach(()=>{
//   console.log('before each calling!')
// });

// afterAll(()=>{
//   console.log('after all calling!')
// });
// afterEach(()=>{
//   console.log('after each calling!')
// });

test('Snapshot for component', () => {
  const { container } = render(<App />); // Destructure the container from the result
  expect(container).toMatchSnapshot();   // Snapshot the container
});

test("Click func test: 01",()=>{
  render(<App/>);
  const btn = screen.getByRole('button');
  fireEvent.click(btn);

  expect(screen.getByText('Update Data')).toBeInTheDocument();
});
test("Click func test: 02",()=>{
  render(<App/>);
  const btn = screen.getByRole('button');
  fireEvent.click(btn);

  expect(screen.getByText('Update Data')).toBeInTheDocument();
});

