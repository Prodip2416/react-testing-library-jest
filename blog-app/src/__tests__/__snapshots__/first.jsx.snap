// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for component 1`] = `
<div>
  <div
    class="App"
  >
    <button>
      Update Data
    </button>
    <p />
  </div>
</div>
`;
